src/PhpParser/Printer/FormatPerservingPrinter.php

```diff
@@ -56,4 +53,12 @@ public function printParsedStmstAndTokens(File $file): string
             $file->getOldTokens()
         );
     }
+
+    public function dumpFile(string $filePath, string $newContent): void
+    {
+        $this->filesystem->dumpFile($filePath, $newContent);
+
+        // @todo how to keep origianl access rights without the SplFileInfo
```

Comment:
debug leftover?

Reply:
it was the original code, I think @TomasVotruba was trying to make not rely on `Filesystem` with different implementation and mark as todo :). 

I just moved the functionality as used to re-use on rollback original file content.

Reply:
I think there was also same issue that file was reprinted always, but with different than orignal rights.
Now it's gone I assume as nobody filled an issue in this absence :)

Reply:
@TomasVotruba do you think commented code can be removed?

Reply:
@staabm @TomasVotruba I removed commented code https://github.com/rectorphp/rector-src/pull/3412/commits/c3a76176710d97e517f7d85c61a9566b4f0095a4 👍